ledger/ledgercore/statedelta.go

```diff
@@ -219,6 +219,32 @@ func (sd *StateDelta) PopulateStateDelta(hdr *bookkeeping.BlockHeader, prevTimes
 	sd.PrevTimestamp = prevTimestamp
 }
 
+// Hydrate reverses the effects of Dehydrate, restoring internal data.
+func (sd *StateDelta) Hydrate() {
+	sd.Accts.Hydrate()
+}
+
+// Dehydrate normalized the fields of this StateDelta, and clears any redundant internal caching.
+// This is useful for comparing StateDelta objects for equality.
+//
+// NOTE: initialHint is lost in dehydration. All other fields can be restored by calling Hydrate()
+func (sd *StateDelta) Dehydrate() {
```

Comment:
Is this code only used in tests? Can we put it in a `_test.go` to ensure it stays that way?

Reply:
Currently yes, only used in tests. The problem is tests from other packages use this, so I'm not sure if I can hide it in a `_test.go` file

Reply:
No, you're right, you won't be able to do that.